packages/mold/package.desc

```diff
@@ -0,0 +1,4 @@
+repository='git https://github.com/rui314/mold.git'
+mirrors='https://github.com/rui314/mold/archive/refs/tags'
+archive_filename='v@{version}'
```

Comment:
Just a comment I realized, this will download v2.31.0.tar.gz into the src directory, which potentially could collide with another package over time.

Ideally there should be a rename step from download filename to locally stored source, here ideally rename to mold-v2.31.0.tar.gz.

But this is github.com automatic generated artifacts I assume, we have the same problem in other packages already.

Reply:
Actually we get away with it because the packages we happen to use that are hosted on github all make use of the `releases` endpoint instead of the `archive` endpoint (basically they all happen to do releases with uploaded artifacts instead of just the auto-generated stuff).

buildroot handles this with a helper that renames the tarball as part of the download. We'd probably need to do something similar but this is the first time we've actually needed it.

Reply:
There is at least one more `packages/duma/package.desc`, but with a slightly different naming.

But so far no chance of collision.